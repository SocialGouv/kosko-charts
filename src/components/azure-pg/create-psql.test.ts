import { createPsqlJob } from "./create-psql.job";

process.env.CI_COMMIT_SHORT_SHA = "b123a99";

test("should create a psql job", () => {
  expect(
    createPsqlJob({
      database: "some-db",
      script: "SELECT VERSION();",
      secretRefName: "some-secret",
    })
  ).toMatchSnapshot();
});

test("should use admin secret as default", () => {
  expect(
    createPsqlJob({
      database: "some-db2",
      script: "SELECT VERSION();",
    })
  ).toMatchSnapshot();
});
