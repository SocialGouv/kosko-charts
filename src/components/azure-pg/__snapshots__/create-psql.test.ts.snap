// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should create a psql job 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {},
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "psql",
              "--dbname",
              "some-db",
              "-c",
              "SELECT VERSION();",
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "some-secret",
                },
              },
            ],
            "image": "postgres:10",
            "imagePullPolicy": "IfNotPresent",
            "name": "psql-job",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;

exports[`should use admin secret as default 1`] = `
Object {
  "apiVersion": "batch/v1",
  "kind": "Job",
  "metadata": Object {},
  "spec": Object {
    "backoffLimit": 0,
    "template": Object {
      "spec": Object {
        "containers": Array [
          Object {
            "command": Array [
              "psql",
              "--dbname",
              "some-db2",
              "-c",
              "SELECT VERSION();",
            ],
            "envFrom": Array [
              Object {
                "secretRef": Object {
                  "name": "azure-pg-admin-user",
                },
              },
            ],
            "image": "postgres:10",
            "imagePullPolicy": "IfNotPresent",
            "name": "psql-job",
            "resources": Object {
              "limits": Object {
                "cpu": "300m",
                "memory": "256Mi",
              },
              "requests": Object {
                "cpu": "100m",
                "memory": "64Mi",
              },
            },
          },
        ],
        "restartPolicy": "Never",
      },
    },
    "ttlSecondsAfterFinished": 86400,
  },
}
`;
